migrations/sql/V2019.04.25.11.30__add_variance_applications_metadata.sql

```diff
@@ -0,0 +1,105 @@
+CREATE TABLE variance_application_status_code (
+    variance_application_status_code character varying(3)                   NOT NULL,
+    description                      character varying(100)                 NOT NULL,
+    active_ind                       boolean                  DEFAULT true  NOT NULL,
+    create_user                      character varying(60)                  NOT NULL,
+    create_timestamp                 timestamp with time zone DEFAULT now() NOT NULL,
+    update_user                      character varying(60)                  NOT NULL,
+    update_timestamp                 timestamp with time zone DEFAULT now() NOT NULL,
+
+    UNIQUE(variance_application_status_code)
+);
+
+ALTER TABLE variance_application_status_code OWNER TO mds;
+
+COMMENT ON TABLE variance_application_status_code IS 'The valid options for a variance / variance application.';
+
+INSERT INTO variance_application_status_code (
+    variance_application_status_code,
+    description,
+    create_user,
+    update_user
+)
+VALUES
+    ('REV', 'In Review', 'system-mds', 'system-mds'),
+    ('NAP', 'Not Applicable', 'system-mds', 'system-mds'),
+    ('APP', 'Approved', 'system-mds', 'system-mds'),
+    ('DEN', 'Denied', 'system-mds', 'system-mds');
+
+
+ALTER TABLE variance
+ADD COLUMN variance_application_status_code character varying(3) DEFAULT 'REV' NOT NULL,
+ADD COLUMN ohsc_ind                         boolean              DEFAULT FALSE NOT NULL,
+ADD COLUMN union_ind                        boolean              DEFAULT FALSE NOT NULL,
+ADD COLUMN inspector_id                     integer                                    ,
+ADD COLUMN variance_no                      integer                            NOT NULL;
+
+COMMENT ON COLUMN variance.variance_no IS 'A public, unique identifier for a mine variance.';
+
+ALTER TABLE variance
+ALTER COLUMN issue_date  DROP NOT NULL,
+ALTER COLUMN expiry_date DROP NOT NULL;
+
+ALTER TABLE variance
+ALTER COLUMN issue_date    DROP DEFAULT,
+ALTER COLUMN expiry_date   DROP DEFAULT,
+ALTER COLUMN received_date DROP DEFAULT;
+
+ALTER TABLE variance
+ADD UNIQUE (mine_guid, variance_no);
+
+ALTER TABLE variance
+ADD CONSTRAINT variance_status_variance_application_status_code_fkey
+    FOREIGN KEY (variance_application_status_code)
+    REFERENCES variance_application_status_code(variance_application_status_code)
+    DEFERRABLE INITIALLY DEFERRED,
+ADD CONSTRAINT variance_inspector_core_user_fkey
+    FOREIGN KEY (inspector_id) REFERENCES core_user(core_user_id) DEFERRABLE INITIALLY DEFERRED,
+
+-- Business Rule Constraints
```

Comment:
are we building these business rules at the db level? I thought we wanted to keep this in the python.

Reply:
This seems appropriate to me given that the applications and variances exist in the same table. It's a few small rules strictly pertaining to the presence/absence of certain values to maintain data integrity.

What are your concerns with this implementation?

Reply:
If they need different rules, then it feels like they should be different tables.... I'm not strongly against it, but we have no other constraints like this in the system, so just poking abit.... 

In theory we could write these as python @validates methods.... which are much more testable and we control them, we are currently running into hiccups when trying to catch db errors globally. @JustinMacaulay do we think we are going to able to do that?

Reply:
Well, we should have the model-level validation anyway. Relying on database errors would be a bad pattern, in my opinion.

We have `NOT NULL` constraints in the database for a reason. These are just `NOT NULL` for an approved variance.

Reply:
Okay. I'll concede